cnctcli/api/products.py

```diff
@@ -3,164 +3,91 @@
-
-    handle_http_error(res)
-
-
-def get_item(endpoint, api_key, product_id, item_id):
-    res = requests.get(
-        f'{endpoint}/products/{product_id}/items/{item_id}',
-        headers=get_headers(api_key),
-    )
-
-    if res.status_code == 200:
-        return res.json()
-
-    if res.status_code == 404:
-        return
-
-    handle_http_error(res)
+def get_item(client, product_id, item_id):
+    try:
+        res = (
+            client.products[product_id]
+            .items[item_id]
+            .get()
+        )
+    except ClientError as error:
+        if error.status_code == 404:
+            return
+        handle_http_error(error)
+    return res
 
 
-def get_item_by_mpn(endpoint, api_key, product_id, mpn):
-    res = requests.get(
-        f'{endpoint}/products/{product_id}/items?eq(mpn,{mpn})',
-        headers=get_headers(api_key),
-    )
+def get_item_by_mpn(client, product_id, mpn):
+    rql = R().mpn.eq(mpn)
 
-    if res.status_code == 200:
-        results = res.json()
-        return results[0] if results else None
+    try:
+        res = (
+            client.products[product_id]
+            .items
+            .filter(rql)
+        )
 
-    if res.status_code == 404:
-        return
+    except ClientError as error:
+        if error.status_code == 404:
+            return
+        handle_http_error(error)
 
-    handle_http_error(res)
+    return res.first()
 
 
-def create_item(endpoint, api_key, product_id, data):
-    res = requests.post(
-        f'{endpoint}/products/{product_id}/items',
-        headers=get_headers(api_key),
-        json=data,
-    )
-    if res.status_code == 201:
-        return res.json()
+def create_item(client, product_id, data):
+    try:
+        res = (
+            client.products[product_id]
+            .items
+            .create(data)
+        )
+    except ClientError as error:
+        handle_http_error(error)
 
-    handle_http_error(res)
+    return res
 
 
-def update_item(endpoint, api_key, product_id, item_id, data):
-    res = requests.put(
-        f'{endpoint}/products/{product_id}/items/{item_id}',
-        headers=get_headers(api_key),
-        json=data,
-    )
-    if res.status_code != 200:
-        handle_http_error(res)
+def update_item(client, product_id, item_id, data):
+    try:
+        res = (
+            client.products[product_id]
+            .items[item_id]
+            .update(data)
+        )
+    except ClientError as error:
+        handle_http_error(error)
 
-    return res.json()
+    return res
 
 
-def delete_item(endpoint, api_key, product_id, item_id):
-    res = requests.delete(
-        f'{endpoint}/products/{product_id}/items/{item_id}',
-        headers=get_headers(api_key),
-    )
-    if res.status_code != 204:
-        handle_http_error(res)
+def delete_item(client, product_id, item_id):
+    try:
+        (
+            client.products[product_id]
+            .items
+            .delete(item_id)
+        )
```

Comment:
this doesn't work.

Should be: client.products[product_id].items[item_id].delete()